http4s/src/main/scala/Http4sConsul.scala

```diff
@@ -62,20 +66,118 @@ final class Http4sConsulClient[F[_]](
   def addCreds(req: Request[F]): Request[F] =
     credentials.fold(req){case (un,pw) => req.putHeaders(Authorization(BasicCredentials(un,pw)))}
 
-  def kvGet(key: Key): F[Option[String]] = {
+  /** A nice place to store the Consul response headers so we can pass them around */
+  case class ConsulHeaders(
+    index:       Long,
+    lastContact: Long,
+    knownLeader: Boolean
+  )
+
+  /** Helper function to get the value of a header out of a Response */
+  def extractHeaderValue(header: String, response: Response[F]): F[String] = {
+    response.headers.get(header.ci) match {
+      case Some(header) => F.pure(header.value)
+      case None         => F.pure(new RuntimeException(s"Header not present in response: $header")).flatMap(F.raiseError)
```

Comment:
Can just `F.raiseError` directly here, if we leave it exceptional.  And I might call it a `NoSuchElementException`.